src/ranged.cpp

```diff
@@ -617,11 +563,12 @@ int player::fire_gun( const tripoint &target, int shots, item& gun )
         }
     }
 
-    practice( skill_gun, train_skill ? 15 : 0 );
-
     // Use different amounts of time depending on the type of gun and our skill
     moves -= time_to_fire( *this, *gun.type );
 
+    practice( skill_gun, xp * ( get_skill_level( skill_gun ) + 1 ) );
+    practice( gun.gun_skill(), dmg );
```

Comment:
Scaling xp gain with damage was considered earlier. It has some issues:
* Bows, crossbows, slings and launchers train horribly slowly
* Shotguns train like mad
* Armored targets don't train skills as well
* Hard targets (currently don't exist, but this just moves problems to the future) don't train skills as well as easy ones
* Since it doesn't depend on range, the optimal way to train skill at accurate ranged shooting becomes shooting everything at point blank. Then training marksmanship explicitly by sniping some distant targets. Nothing in between.
* Short term realism problems. That is, even if in the end it results in a realistic situation, it doesn't make much sense when we consider the current situation (shot big bullet from accurate gun - gain more skill).